lib_eio_luv/eio_luv.ml

```diff
@@ -475,6 +475,43 @@ let luv_ip_addr_to_eio addr =
   let port = Luv.Sockaddr.port addr |> Option.get in
   (Eio_unix.Ipaddr.of_unix (Unix.inet_addr_of_string host), port)
 
+module Endpoint = struct
+  type 'a t = [`UDP] Handle.t
+
+  let recv (sock:'a t) buf =
+    let r = enter (fun t k ->
+        Fibre_context.set_cancel_fn k.fibre (fun ex ->
+            Luv.UDP.recv_stop (Handle.get "recv_into:cancel" sock) |> or_raise;
+            enqueue_failed_thread t k ex
+          );
+        Luv.UDP.recv_start (Handle.get "recv_start" sock) ~allocate:(fun _ -> buf) (fun r ->
+            Luv.UDP.recv_stop (Handle.get "recv_stop" sock) |> or_raise;
+            if Fibre_context.clear_cancel_fn k.fibre then enqueue_thread t k r
+          )
+      ) in
+    match r with
```

Comment:
The [Luv docs](https://aantron.github.io/luv/luv/Luv/UDP/index.html#val-recv_start) say:
> None usually indicates EAGAIN in libuv; libuv still calls the callback, in order to give the C user a chance to deallocate the data buffer. Since this is not usually an issue in OCaml, it is usually safe to simply ignore calls to the callback with sender address None.

Reply:
Yep I saw that. What's the best way to handle `EAGAIN`? Just recursively call `recv` again ?

Reply:
Just don't do anything in that case I think, not even `recv_stop`.